io_export_objex2/interface.py

```diff
@@ -1399,140 +1423,345 @@ def draw_optional(prop_enable, prop_data):
-                box.prop_search(data, 'branch_to_object', bpy.data, 'objects')
-                box.label(text='(mesh objects only)')
+        
+        self.menu_material(context)
+    
+    def get_icon(self, attr):
+        if attr == True:
+            return 'DOWNARROW_HLT'
+        else:
+            return 'RIGHTARROW'
+
+    def menu_collision(self, context):
+        objex_scene = context.scene.objex_bonus
+        material = context.material
+        data = material.objex_bonus
+
+        box = self.layout.box()
+        box.use_property_split = False
+        row = box.row()
+        row.alignment = 'CENTER'
+        row.label(text='Collision')
+
+        col = data.collision
+
+        box.prop(col, 'sound')
+        box.prop(col, 'floor')
+        box.prop(col, 'wall')
+        box.prop(col, 'special')
+
+        row = box.row()
+        row.prop(col, 'ignore_camera')
+        row.prop(col, 'ignore_entity')
+
+        row = box.row()
+        row.prop(col, 'ignore_ammo')
+        row.prop(col, 'horse')
+
+        row = box.row()
+        row.prop(col, 'one_lower')
+        row.prop(col, 'wall_damage')
+
+        row = box.row()
+        row.prop(col, 'hookshot')
+        row.prop(col, 'steep')
+
+        def draw_optional(prop_enable, prop_data):
+            if getattr(col, prop_enable):
+                row = box.row()
+                row.prop(col, prop_enable, text='')
+                row.prop(col, prop_data)
             else:
-                self.layout.prop(data, 'branch_to_object')
-            if data.branch_to_object: # branch_to_object is a MESH object
+                row = box.row()
+                row.prop(col, prop_enable, text='')
+                column = row.column()
+                column.enabled = False
+                column.prop(col, prop_data)
+
+        draw_optional('warp_enabled', 'warp_exit_index')
+        draw_optional('camera_enabled', 'camera_index')
+        draw_optional('echo_enabled', 'echo_index')
+        draw_optional('lighting_enabled', 'lighting_index')
+
+        if col.conveyor_enabled:
+            box.prop(col, 'conveyor_enabled')
+            sub_box = box.box()
+            sub_box.prop(col, 'conveyor_direction')
+            sub_box.row().prop(col, 'conveyor_speed', expand=True)
+            sub_box.prop(col, 'conveyor_inherit')
+        else:
+            box.prop(col, 'conveyor_enabled')
+
+    def menu_material(self, context):
+        material = context.material
+        data = material.objex_bonus
+        objex_scene = context.scene.objex_bonus
+        mode_menu = objex_scene.mode_menu
+
+        data_is_empty = False
+        if data.empty or material.name.startswith('empty.'):
+            data_is_empty = True
+
+        box = self.layout.box()
+        box.use_property_split = False
+        box.prop(objex_scene, 'menu_tools', icon=self.get_icon(getattr(objex_scene, 'menu_tools')), emboss=False)
+        if objex_scene.menu_tools == True:
```

Comment:
```suggestion
        if objex_scene.menu_tools:
```
and anywhere else you have `== True` Madge